datadog/fwprovider/resource_datadog_integration_gcp.go

```diff
@@ -341,12 +348,22 @@ func (r *integrationGcpResource) addRequiredFieldsToBody(body *datadogV1.GCPAcco
 	body.SetPrivateKeyId(state.PrivateKeyId.ValueString())
 }
 
-func (r *integrationGcpResource) addOptionalFieldsToBody(body *datadogV1.GCPAccount, state integrationGcpModel) {
+func (r *integrationGcpResource) addOptionalFieldsToBody(ctx context.Context, body *datadogV1.GCPAccount, state integrationGcpModel) diag.Diagnostics {
+	diags := diag.Diagnostics{}
 	body.SetAutomute(state.Automute.ValueBool())
 	body.SetIsCspmEnabled(state.CspmResourceCollectionEnabled.ValueBool())
 	body.SetIsSecurityCommandCenterEnabled(state.IsSecurityCommandCenterEnabled.ValueBool())
 	body.SetHostFilters(state.HostFilters.ValueString())
+
+	runFilters := make([]string, 0)
+	if !state.CloudRunRevisionFilters.IsNull() {
+		diags.Append(state.CloudRunRevisionFilters.ElementsAs(ctx, &runFilters, false)...)
+	}
+	body.SetCloudRunRevisionFilters(runFilters)
```

Comment:
Is this safe? Won't this send an empty array even if the state does not have filters set? FYI the dogweb v2 API takes whatever you give it and sets it. So, if you send an empty array, you'll overwrite any existing filters.

Reply:
Spoke offline. I think we're good here. If `cloud_run_revision_filters` is absent from the terraform object, then we should delete them. 👍